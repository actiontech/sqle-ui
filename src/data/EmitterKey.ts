enum EmitterKey {
  Refresh_User_list = 'REFRESH_USER_LIST',
  Refresh_Role_list = 'REFRESH_ROLE_LIST',
  Refresh_Whitelist_List = 'REFRESH_WHITELIST_LIST',

  Refresh_Rule_Template_List = 'REFRESH_RULE_TEMPLATE_LIST',

  Reset_Create_Order_Form = 'RESET_CREATE_ORDER_FORM',

  Reset_Test_Data_Source_Connect = 'RESET_TEST_DATA_SOURCE_CONNECT',

  Reset_Workflow_Template_Form = 'RESET_WORKFLOW_TEMPLATE_FORM',

  Rest_Audit_Plan_Form = 'REST_AUDIT_PLAN_FORM',

  Refresh_Audit_Plan_Record = 'REFRESH_AUDIT_PLAN_RECORD',
}

export default EmitterKey;
