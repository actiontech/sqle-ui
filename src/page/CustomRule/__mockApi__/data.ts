import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
