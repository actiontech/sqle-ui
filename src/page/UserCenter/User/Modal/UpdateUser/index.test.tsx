import {
  cleanup,
  fireEvent,
  render,
  waitFor,
  screen,
} from '@testing-library/react';
import UpdateUser from '.';
import user from '../../../../../api/user';
import EmitterKey from '../../../../../data/EmitterKey';
import { ModalName } from '../../../../../data/ModalName';
import {
  mockUseDispatch,
  mockUseSelector,
} from '../../../../../testUtils/mockRedux';
import {
  mockUseRole,
  mockUseUserGroup,
  resolveThreeSecond,
} from '../../../../../testUtils/mockRequest';
import EventEmitter from '../../../../../utils/EventEmitter';

describe('User/Modal/AddUser', () => {
  let userRoleSpy: jest.SpyInstance;
  let useUserGroupSpy: jest.SpyInstance;
  let dispatchSpy: jest.Mock;

  beforeEach(() => {
    userRoleSpy = mockUseRole();
    useUserGroupSpy = mockUseUserGroup();
    mockUseSelector({
      userManage: {
        modalStatus: { [ModalName.Update_User]: true },
        selectUser: {
          user_name: 'root',
          email: 'user@123.com',
          role_name_list: ['role_name1'],
          user_group_name_list: ['user_group_name1'],
          wechat_id: '11231123',
        },
      },
    });
    const { scopeDispatch } = mockUseDispatch();
    dispatchSpy = scopeDispatch;
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  test('should get instance list and username list when modal is opened', async () => {
    render(<UpdateUser />);
    expect(userRoleSpy).toBeCalledTimes(1);
    expect(useUserGroupSpy).toBeCalledTimes(1);
    cleanup();
    userRoleSpy.mockClear();
    useUserGroupSpy.mockClear();
    mockUseSelector({
      userManage: { modalStatus: { [ModalName.Update_User]: false } },
    });
    render(<UpdateUser />);
    expect(userRoleSpy).not.toBeCalled();
    expect(useUserGroupSpy).not.toBeCalled();
  });

  test('should send update user request when user click submit button', async () => {
    render(<UpdateUser />);
    const updateUserSpy = jest.spyOn(user, 'updateUserV1');
    updateUserSpy.mockImplementation(() => resolveThreeSecond({}));
    const emitSpy = jest.spyOn(EventEmitter, 'emit');
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(screen.getByLabelText('user.userForm.username')).toHaveValue('root');
    expect(screen.getByLabelText('user.userForm.username')).toHaveAttribute(
      'disabled'
    );
    expect(
      screen.queryByLabelText('user.userForm.password')
    ).not.toBeInTheDocument();

    fireEvent.input(screen.getByLabelText('user.userForm.email'), {
      target: { value: 'newuser@163.com' },
    });

    fireEvent.click(screen.getByText('user.userForm.disabled'));

    fireEvent.click(screen.getByText('common.submit'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(screen.getByText('common.submit').parentNode).toHaveClass(
      'ant-btn-loading'
    );
    expect(screen.getByText('common.close').parentNode).toHaveAttribute(
      'disabled'
    );
    expect(updateUserSpy).toBeCalledTimes(1);
    expect(updateUserSpy).toBeCalledWith({
      email: 'newuser@163.com',
      user_name: 'root',
      is_disabled: true,
      role_name_list: ['role_name1'],
      user_group_name_list: ['user_group_name1'],
      wechat_id: '11231123',
    });
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(
      screen.queryByText('user.updateUser.updateSuccessTips')
    ).toBeInTheDocument();
    expect(emitSpy).toBeCalledTimes(1);
    expect(emitSpy).toBeCalledWith(EmitterKey.Refresh_User_list);
    expect(screen.getByLabelText('user.userForm.username')).toHaveValue('');
    expect(dispatchSpy).toBeCalledTimes(1);
    expect(dispatchSpy).toBeCalledWith({
      payload: {
        modalName: 'Update_User',
        status: false,
      },
      type: 'user/updateModalStatus',
    });
    expect(screen.getByText('common.submit').parentNode).not.toHaveClass(
      'ant-btn-loading'
    );
    expect(screen.getByText('common.close').parentNode).not.toHaveAttribute(
      'disabled'
    );
  });

  test('should send update user request when the username is any value', async () => {
    userRoleSpy.mockClear();
    mockUseSelector({
      userManage: {
        modalStatus: { [ModalName.Update_User]: true },
        selectUser: {
          user_name: 'san.zhang',
          email: 'user@123.com',
          role_name_list: ['role_name1'],
        },
      },
    });
    render(<UpdateUser />);
    expect(userRoleSpy).toBeCalledTimes(1);

    const updateUserSpy = jest.spyOn(user, 'updateUserV1');
    updateUserSpy.mockImplementation(() => resolveThreeSecond({}));
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(screen.getByLabelText('user.userForm.username')).toHaveValue(
      'san.zhang'
    );

    fireEvent.click(screen.getByText('common.submit'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(userRoleSpy).toBeCalledTimes(1);
    expect(updateUserSpy).toBeCalledWith({
      email: 'user@123.com',
      user_name: 'san.zhang',
      is_disabled: false,
      role_name_list: ['role_name1'],
      user_group_name_list: [],
    });
  });

  test('should not set disable filed when user name is admin', async () => {
    userRoleSpy.mockClear();
    mockUseSelector({
      userManage: {
        modalStatus: { [ModalName.Update_User]: true },
        selectUser: {
          user_name: 'admin',
          email: 'user@123.com',
          role_name_list: ['role_name1'],
        },
      },
    });
    render(<UpdateUser />);
    expect(userRoleSpy).toBeCalledTimes(1);

    const updateUserSpy = jest.spyOn(user, 'updateUserV1');
    updateUserSpy.mockImplementation(() => resolveThreeSecond({}));
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(
      screen.queryByLabelText('user.userForm.disabled')
    ).not.toBeInTheDocument();

    fireEvent.click(screen.getByText('common.submit'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(userRoleSpy).toBeCalledTimes(1);
    expect(updateUserSpy).toBeCalledWith({
      email: 'user@123.com',
      user_name: 'admin',
      role_name_list: ['role_name1'],
      user_group_name_list: [],
    });
  });

  test('should send update user request when user clear email', async () => {
    userRoleSpy.mockClear();
    mockUseSelector({
      userManage: {
        modalStatus: { [ModalName.Update_User]: true },
        selectUser: {
          user_name: 'san.zhang',
          email: 'user@123.com',
          role_name_list: ['role_name1'],
        },
      },
    });
    render(<UpdateUser />);
    expect(userRoleSpy).toBeCalledTimes(1);

    const updateUserSpy = jest.spyOn(user, 'updateUserV1');
    updateUserSpy.mockImplementation(() => resolveThreeSecond({}));
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(screen.getByLabelText('user.userForm.username')).toHaveValue(
      'san.zhang'
    );

    fireEvent.input(screen.getByLabelText('user.userForm.email'), {
      target: { value: '' },
    });

    fireEvent.click(screen.getByText('common.submit'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(userRoleSpy).toBeCalledTimes(1);
    expect(updateUserSpy).toBeCalledWith({
      email: '',
      user_name: 'san.zhang',
      is_disabled: false,
      role_name_list: ['role_name1'],
      user_group_name_list: [],
    });
  });
});
