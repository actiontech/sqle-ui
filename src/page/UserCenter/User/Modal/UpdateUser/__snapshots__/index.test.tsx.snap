// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User/Modal/UpdateUser should match snapshot 1`] = `
<body
  class="ant-scrolling-effect"
  style="width: calc(100% - 0px); overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
>
  <div />
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        aria-labelledby="rcDialogTitle0"
        class="ant-modal-wrap"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="ant-modal"
          role="document"
          style="width: 520px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="rcDialogTitle0"
              >
                user.updateUser.modalTitle
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal"
              >
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class="ant-form-item-required"
                      for="username"
                      title="user.userForm.username"
                    >
                      user.userForm.username
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input ant-input-disabled"
                          disabled=""
                          id="username"
                          placeholder="common.form.placeholder.input"
                          type="text"
                          value="root"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="email"
                      title="user.userForm.email"
                    >
                      user.userForm.email
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="email"
                          placeholder="common.form.placeholder.input"
                          type="text"
                          value="user@123.com"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item input-number-hide-arrows ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="phone"
                      title="user.userForm.phone"
                    >
                      user.userForm.phone
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="phone"
                          placeholder="common.form.placeholder.input"
                          type="number"
                          value="13312341234"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="wechat"
                      title="user.userForm.wechat"
                    >
                      user.userForm.wechat
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="wechat"
                          placeholder="common.form.placeholder.input"
                          type="text"
                          value="11231123"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="managementPermissionCodeList"
                      title="user.userForm.managerPermission"
                    >
                      user.userForm.managerPermission
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-select ant-select-multiple ant-select-show-search"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <div
                              class="ant-select-selection-overflow"
                            >
                              <div
                                class="ant-select-selection-overflow-item"
                                style="opacity: 1;"
                              >
                                <span
                                  class="ant-select-selection-item"
                                >
                                  <span
                                    class="ant-select-selection-item-content"
                                  >
                                    创建项目
                                  </span>
                                  <span
                                    aria-hidden="true"
                                    class="ant-select-selection-item-remove"
                                    style="user-select: none;"
                                    unselectable="on"
                                  >
                                    <span
                                      aria-label="close"
                                      class="anticon anticon-close"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="close"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                              <div
                                class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                                style="opacity: 1;"
                              >
                                <div
                                  class="ant-select-selection-search"
                                  style="width: 0px;"
                                >
                                  <input
                                    aria-activedescendant="managementPermissionCodeList_list_0"
                                    aria-autocomplete="list"
                                    aria-controls="managementPermissionCodeList_list"
                                    aria-haspopup="listbox"
                                    aria-owns="managementPermissionCodeList_list"
                                    autocomplete="off"
                                    class="ant-select-selection-search-input"
                                    id="managementPermissionCodeList"
                                    readonly=""
                                    role="combobox"
                                    style="opacity: 0;"
                                    type="search"
                                    unselectable="on"
                                    value=""
                                  />
                                  <span
                                    aria-hidden="true"
                                    class="ant-select-selection-search-mirror"
                                  >
                                    
                                     
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="userGroupList"
                      title="user.userForm.userGroup"
                    >
                      user.userForm.userGroup
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-select ant-select-multiple ant-select-show-search"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <div
                              class="ant-select-selection-overflow"
                            >
                              <div
                                class="ant-select-selection-overflow-item"
                                style="opacity: 1;"
                              >
                                <span
                                  class="ant-select-selection-item"
                                >
                                  <span
                                    class="ant-select-selection-item-content"
                                  >
                                    user_group_name1
                                  </span>
                                  <span
                                    aria-hidden="true"
                                    class="ant-select-selection-item-remove"
                                    style="user-select: none;"
                                    unselectable="on"
                                  >
                                    <span
                                      aria-label="close"
                                      class="anticon anticon-close"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="close"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                              <div
                                class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                                style="opacity: 1;"
                              >
                                <div
                                  class="ant-select-selection-search"
                                  style="width: 0px;"
                                >
                                  <input
                                    aria-activedescendant="userGroupList_list_0"
                                    aria-autocomplete="list"
                                    aria-controls="userGroupList_list"
                                    aria-haspopup="listbox"
                                    aria-owns="userGroupList_list"
                                    autocomplete="off"
                                    class="ant-select-selection-search-input"
                                    id="userGroupList"
                                    readonly=""
                                    role="combobox"
                                    style="opacity: 0;"
                                    type="search"
                                    unselectable="on"
                                    value=""
                                  />
                                  <span
                                    aria-hidden="true"
                                    class="ant-select-selection-search-mirror"
                                  >
                                    
                                     
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="disabled"
                      title="user.userForm.disabled"
                    >
                      user.userForm.disabled
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          aria-checked="false"
                          class="ant-switch"
                          id="disabled"
                          role="switch"
                          type="button"
                        >
                          <div
                            class="ant-switch-handle"
                          />
                          <span
                            class="ant-switch-inner"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn"
                type="button"
              >
                <span>
                  common.close
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  common.submit
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`User/Modal/UpdateUser should match snapshot 2`] = `
<body
  class="ant-scrolling-effect"
  style="width: calc(100% - 0px); overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
>
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        aria-labelledby="rcDialogTitle1"
        class="ant-modal-wrap"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="ant-modal"
          role="document"
          style="width: 520px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="rcDialogTitle1"
              >
                user.updateUser.modalTitle
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal"
              >
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class="ant-form-item-required"
                      for="username"
                      title="user.userForm.username"
                    >
                      user.userForm.username
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input ant-input-disabled"
                          disabled=""
                          id="username"
                          placeholder="common.form.placeholder.input"
                          type="text"
                          value="root"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="email"
                      title="user.userForm.email"
                    >
                      user.userForm.email
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="email"
                          placeholder="common.form.placeholder.input"
                          type="text"
                          value="user@123.com"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item input-number-hide-arrows ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="phone"
                      title="user.userForm.phone"
                    >
                      user.userForm.phone
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="phone"
                          placeholder="common.form.placeholder.input"
                          type="number"
                          value="13312341234"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="wechat"
                      title="user.userForm.wechat"
                    >
                      user.userForm.wechat
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="wechat"
                          placeholder="common.form.placeholder.input"
                          type="text"
                          value="11231123"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="managementPermissionCodeList"
                      title="user.userForm.managerPermission"
                    >
                      user.userForm.managerPermission
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-select ant-select-multiple ant-select-show-search"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <div
                              class="ant-select-selection-overflow"
                            >
                              <div
                                class="ant-select-selection-overflow-item"
                                style="opacity: 1;"
                              >
                                <span
                                  class="ant-select-selection-item"
                                >
                                  <span
                                    class="ant-select-selection-item-content"
                                  >
                                    1
                                  </span>
                                  <span
                                    aria-hidden="true"
                                    class="ant-select-selection-item-remove"
                                    style="user-select: none;"
                                    unselectable="on"
                                  >
                                    <span
                                      aria-label="close"
                                      class="anticon anticon-close"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="close"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                              <div
                                class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                                style="opacity: 1;"
                              >
                                <div
                                  class="ant-select-selection-search"
                                  style="width: 0px;"
                                >
                                  <input
                                    aria-activedescendant="managementPermissionCodeList_list_0"
                                    aria-autocomplete="list"
                                    aria-controls="managementPermissionCodeList_list"
                                    aria-haspopup="listbox"
                                    aria-owns="managementPermissionCodeList_list"
                                    autocomplete="off"
                                    class="ant-select-selection-search-input"
                                    id="managementPermissionCodeList"
                                    readonly=""
                                    role="combobox"
                                    style="opacity: 0;"
                                    type="search"
                                    unselectable="on"
                                    value=""
                                  />
                                  <span
                                    aria-hidden="true"
                                    class="ant-select-selection-search-mirror"
                                  >
                                    
                                     
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="userGroupList"
                      title="user.userForm.userGroup"
                    >
                      user.userForm.userGroup
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-select ant-select-multiple ant-select-show-search"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <div
                              class="ant-select-selection-overflow"
                            >
                              <div
                                class="ant-select-selection-overflow-item"
                                style="opacity: 1;"
                              >
                                <span
                                  class="ant-select-selection-item"
                                >
                                  <span
                                    class="ant-select-selection-item-content"
                                  >
                                    user_group_name1
                                  </span>
                                  <span
                                    aria-hidden="true"
                                    class="ant-select-selection-item-remove"
                                    style="user-select: none;"
                                    unselectable="on"
                                  >
                                    <span
                                      aria-label="close"
                                      class="anticon anticon-close"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="close"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                              <div
                                class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                                style="opacity: 1;"
                              >
                                <div
                                  class="ant-select-selection-search"
                                  style="width: 0px;"
                                >
                                  <input
                                    aria-activedescendant="userGroupList_list_0"
                                    aria-autocomplete="list"
                                    aria-controls="userGroupList_list"
                                    aria-haspopup="listbox"
                                    aria-owns="userGroupList_list"
                                    autocomplete="off"
                                    class="ant-select-selection-search-input"
                                    id="userGroupList"
                                    readonly=""
                                    role="combobox"
                                    style="opacity: 0;"
                                    type="search"
                                    unselectable="on"
                                    value=""
                                  />
                                  <span
                                    aria-hidden="true"
                                    class="ant-select-selection-search-mirror"
                                  >
                                    
                                     
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item ant-form-item-has-success"
                >
                  <div
                    class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                  >
                    <label
                      class=""
                      for="disabled"
                      title="user.userForm.disabled"
                    >
                      user.userForm.disabled
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          aria-checked="false"
                          class="ant-switch"
                          id="disabled"
                          role="switch"
                          type="button"
                        >
                          <div
                            class="ant-switch-handle"
                          />
                          <span
                            class="ant-switch-inner"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn"
                type="button"
              >
                <span>
                  common.close
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  common.submit
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
